### ISSUE TITLE
Need for Speed - Most Wanted (Europe) car glitch

### ISSUE THREAD

**Issue by benderscruffy** [TASVideos/desmume#212]
using opengl 3.2 the car is glitchy when it spins
![need](https://user-images.githubusercontent.com/2696404/47612821-4ff60f00-dabd-11e8-9368-620fcbe19cf5.PNG)

**Comment by rogerman** [comment]
Works for me. Please upload a copy of your .ini file so that we can try to reproduce the problem with your settings.
![nfsmw_works_fine](https://user-images.githubusercontent.com/6055122/47613047-a6733700-da44-11e8-8b98-4c82e354aab5.png)

**Comment by benderscruffy** [comment]
ok
[desmume.zip](https://github.com/TASVideos/desmume/files/2522557/desmume.zip)

**Comment by zeromus** [comment]
same problem for me, using all defaults (but change to opengl 3.2)

**Comment by rogerman** [comment]
Okay, so I can confirm that the bug exists. I made the mistake of running some test OpenGL code that was intended to fix the Customize screen of Sands of Destruction (which I thought was unrelated), and that's why Need For Speed rendered correctly in my case. After stashing all my code changes and doing a complete project clean and rebuild, I was able to reproduce the bug as shown in the original post.
![nfsmw_fails](https://user-images.githubusercontent.com/6055122/47678002-eee24000-db7d-11e8-9d98-8edbc885bbbb.png)

In this scene, there are two sets of polygons at play here. The first set of polygons does the usual depth-less-than test to draw the geometry of the car using some generic beige color. The second set of polygons then uses the depth-equals test to draw the paint color onto the car.

The bug is related to stencil buffer issues in the depth-equals test code in OpenGLRenderer::DrawOtherPolygons(). Basically, the depth-equals test code has to do some stencil buffer wrangling to work, and it is precisely this wrangling that is causing polygons to disappear.

Since OpenGLRenderer::DrawOtherPolygons() is a shared method for all OpenGL renderers, this means that this bug will exist for all OpenGL renderers.
